dot/network/discovery_test.go

```diff
@@ -40,12 +41,16 @@ func newTestDiscovery(t *testing.T, num int) []*discovery {
 			NoBootstrap: true,
 			NoMDNS:      true,
 		}
+		dstmpdir := t.TempDir()
+		ds, _ := badger.NewDatastore(path.Join(dstmpdir, "libp2p-datastore"), &badger.DefaultOptions)
```

Comment:
I think you can use an in memory datastore here

Reply:
Done!